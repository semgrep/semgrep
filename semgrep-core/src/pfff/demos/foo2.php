<?php

require_once $_SERVER['PHP_ROOT'].'/lib/foo.php';
require_module('foo/bar');

class Foo extends Bar {
  public function foo() {
    foo_func();
  }

  public static function bar() {
    $bar = new Bar();
    return $bar->bar();
  }
}

function bar_func() {
  return 1;
}
