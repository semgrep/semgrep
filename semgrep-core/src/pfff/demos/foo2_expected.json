{"includes":
  [{"file":"lib/foo.php", "line":3, "column":1},
   {"file":"flib/foo/bar/__init__.php", "line":4, "column":1}],
 "provides":
  [{"class":"Foo", "line":6, "column":1},
   {"func":"bar_func", "line":17, "column":1}],
 "depends":
  [{"func":"require_module", "line":4, "column":1},
   {"class":"Bar", "line":6, "column":18},
   {"func":"foo_func", "line":8, "column":4},
   {"class":"Bar", "line":12, "column":15}]}
